### ISSUE TITLE
Allow limiting the types of modules that are allowed to be added to a given Modular page type

### ISSUE THREAD

**Issue by mikecrittenden** [getgrav/grav#1716]
As a made up example, I would like to have the ability to create a new "Contact" modular page type which will only allow modules of types "Contact Form", "Map", or "Address". Creating other types of modules inside that page would not be allowed in the UI, and if done via creating the folders/files directly then they would be ignored. 

The idea here is that it allows site creators to make sites that are "safer" for clients to edit and more difficult for clients to mess up, by limiting the types of modules that can appear on which types of pages.

Is this a feature that would be considered, or does this kind of go contrary to the types of problems that Modular Pages are built to solve?

**Comment by rhukster** [comment]
So are you thinking that this would be some field in the page header?  Something that contains an array of 'valid' modular page types that you can pick from underneath.  The problem is that this is really only an admin feature as there's really no way to force things for the user when editing files directly.  I think if anything this should be an admin feature, and not a core feature.

**Comment by mikecrittenden** [comment]
Yeah something like that. It seems enforceable in the files directly by
just ignoring any that aren't valid as I mentioned above. Is that not what
you mean?
